pkg/runtime/wfengine/workflow.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package wfengine
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/microsoft/durabletask-go/api"
+	"github.com/microsoft/durabletask-go/backend"
+
+	"github.com/dapr/dapr/pkg/actors"
+)
+
+type workflowActor struct {
+	actorRuntime actors.Actors
+	states       map[string]*workflowState
+	scheduler    workflowScheduler
+}
+
+type workflowState struct {
+	inbox   []*backend.HistoryEvent
+	history []*backend.HistoryEvent
+
+	// runtimeState is a cached copy of the [backend.OrchestrationRuntimeState]
+	runtimeState *backend.OrchestrationRuntimeState
+}
+
+const (
+	CallbackChannelProperty = "dapr.callback"
+
+	CreateWorkflowInstanceMethod = "CreateWorkflowInstance"
+	GetWorkflowMetadataMethod    = "GetWorkflowMetadata"
+
+	createWorkflowReminder = "start" // internal reminder for executing workflows
+)
+
+func NewWorkflowActor(scheduler workflowScheduler) actors.InternalActor {
+	return &workflowActor{
+		states:    make(map[string]*workflowState),
+		scheduler: scheduler,
+	}
+}
+
+// SetActorRuntime implements actors.InternalActor
+func (wf *workflowActor) SetActorRuntime(actorRuntime actors.Actors) {
+	wf.actorRuntime = actorRuntime
+}
+
+// InvokeMethod implements actors.InternalActor
+func (wf *workflowActor) InvokeMethod(ctx context.Context, actorID string, methodName string, request []byte) (interface{}, error) {
+	wfLogger.Debugf("invoking method '%s' on workflow actor '%s'", methodName, actorID)
+
+	var result interface{}
+	var err error
+
+	switch methodName {
+	case CreateWorkflowInstanceMethod:
+		err = wf.createWorkflowInstanceMethod(ctx, actorID, request)
+	case GetWorkflowMetadataMethod:
+		result, err = wf.getWorkflowMetadata(actorID)
+	default:
+		err = fmt.Errorf("no such method: %s", methodName)
+	}
+	return result, err
+}
+
+// InvokeReminder implements actors.InternalActor
+func (wf *workflowActor) InvokeReminder(ctx context.Context, actorID string, reminderName string, params []byte) error {
+	wfLogger.Debugf("invoking reminder '%s' on workflow actor '%s'", reminderName, actorID)
+
+	var err error
+	switch reminderName {
+	case createWorkflowReminder:
```

Comment:
Are we going to eventually have multiple types of reminders? If not, what is the purpose of using the switch case statement here?

Reply:
Originally, I thought there would be separate reminders, but as I continued working on this in a separate branch, I realized that I only need it for this one thing. This `switch` will be removed in a separate PR.